.github/workflows/auto-coderabbit-review-and-update.yml

```
@@ -1,89 +1,76 @@
 jobs:
-  trigger-coderabbit:
-    if: github.event_name == 'pull_request'
+  generate_and_update:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
     steps:
-      - name: Comment to trigger CodeRabbit review
-        uses: actions/github-script@v7
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)